src/interchange/src/json.rs

```diff
@@ -377,16 +381,35 @@ fn build_row_schema_fields(
     columns: &[(ColumnName, ColumnType)],
     type_namer: &mut Namer,
     custom_names: &BTreeMap<GlobalId, String>,
+    set_null_defaults: bool,
 ) -> Vec<serde_json::Value> {
     let mut fields = Vec::new();
     let mut field_namer = Namer::default();
     for (name, typ) in columns.iter() {
         let (name, _seen) = field_namer.valid_name(name.as_str());
-        let field_type = build_row_schema_field(type_namer, custom_names, typ);
-        fields.push(json!({
-            "name": name,
-            "type": field_type,
-        }));
+        let field_type =
+            build_row_schema_field_type(type_namer, custom_names, typ, set_null_defaults);
+
+        let is_nullable_union = field_type.is_array();
```

Comment:
I don't think this is right, we also need to check that the first element is `"null"`. I didn't really understand the argument for making that a soft assert as opposed to part of the actual logic.

Reply:
I interpreted it as that a nullable union is the only kind of union we are currently emitting. And it will force us to update this logic when we do the work for supporting user defaults. 

I don't have strong opinions here (both works IMO, we are either depending upon asserts to make sure logic is as expected or we depend upon good test coverage which should be there anyways).

Reply:
Making it part of the logic instead of asserts does make the code more flexible though.